### DISCUSSION TITLE
[Feature] encryption of online data

### DISCUSSION THREAD

**Discussion by Schweber** [TechbeeAT/jtxBoard#1591] [Ideas]
**Is your feature request related to a problem? Please describe.**

I’m syncing the data of jtx Board via the integration of DAVx5.

It works fine but i’d like jtx Board to only upload encrypted data so my data can’t be read by everyone having access to my davx location.

Interoperability with other apps is not a concern to me as i’m only using jtx board.

**Describe the solution you'd like**

I’d like jtx Board to offer the encryption of all uploaded data via davx with a password. The password should only be prompted once by jtx Board during setup and be automatically applied thereafter (so no prompt each time there is a synchronization).

**Describe alternatives you've considered**

I didn’t find a FLOSS solution that has the capabilities of jtx Board plus encryption.

**Comment by patrickunterwegs** [comment]
Hi @Schweber , 
this is basically a nice idea, but really hard to implement. 
Within jtx Board the data is stored in a database that is only accessible within the app - or if you're using a sync app, the data is accessible for the sync app, but only data that is coming from the given sync app. So within jtx Board and the sync app, there's not really a need for an encryption. 

DAVx5 to the server could maybe use encryption, but on what? The structure of the iCalendar file must be preserved for the exchange with a CalDAV server - there is no standard that would define how such an exchange would take place in an encrypted way. Would then DAVx5 need to encrypt text only? Is it worth the encryption then when rather the communication channel should be secured? How would a key be stored? Does it really add value when a custom encryption is established that only works with a specific app combination?

Those questions are just examples for thoughts that come up when thinking about such an encription. And in any case, I believe DAVx5 is rather the place where the encryption should be done. I'm up for discussion, but I just see it very unlikely that something like this can be done in a meaningful way ;-)

**Reply by Schweber** [reply]
Ok, then i propably have a naive misconception about how jtx board interacts with davx5:

I thought it can say "hey davx5, please sync the following files/data" and davx5 goes along and does that. If this were the case, then full encryption would be possible.

If davx5 needs to actually "understand" the data it is given, then at least the content of the data can be encrypted.

For me it would be sufficient if the content of the files was encrypted. Then i could take notes in jtx board that are stored encrypted online. Metadata, Headers etc. are not a concern for my usecase, so they could stay unencrypted, so that the processing of the files is not impeded. 

Anyway, thank you for your kind response and your work on jtx board!

**Reply by zc-devs** [reply]
If DAVx5 could encrypt content, then it would (?) also work for contacts, which is my second wanted feature :)

**Reply by patrickunterwegs** [reply]
> Ok, then i propably have a naive misconception about how jtx board interacts with davx5:
> 
> I thought it can say "hey davx5, please sync the following files/data" and davx5 goes along and does that. If this were the case, then full encryption would be possible.
> 
> If davx5 needs to actually "understand" the data it is given, then at least the content of the data can be encrypted.
> 
> For me it would be sufficient if the content of the files was encrypted. Then i could take notes in jtx board that are stored encrypted online. Metadata, Headers etc. are not a concern for my usecase, so they could stay unencrypted, so that the processing of the files is not impeded.
> 
> Anyway, thank you for your kind response and your work on jtx board!

Hi @Schweber , sorry for the late reply, I somehow missed your response. 

So yes, DAVx5 needs to understand the data as well as the server itself, as all of it must comply to the iCalendar standard (and is validated on the way). 

Regarding an encryption of the content: 
Well, basically that could be possible. I'm not a big fan of this. Besides the fact that technically it comes with a couple of questions how to properly encrypt and save a passphrase, the thing is that other apps can't make use of the data. This makes the whole open approach of jtx Board (I mean by being compatible with any other app that uses the iCalendar standard for journals, notes and tasks) kind of obsolete - as jtx Board is the only app that could decrypt the data again... So I'm really hesitant to further follow up on this.

**Reply by Schweber** [reply]
I understand that, no worries
